pkg/engine/events.go

```diff
@@ -370,6 +370,10 @@ func filterPropertyMap(propertyMap resource.PropertyMap, debug bool) resource.Pr
 			}
 		case resource.Secret:
 			return "[secret]"
+		case resource.Resource:
+			return resource.Resource{
+				Urn: filterPropertyValue(t.Urn),
```

Comment:
this is probably for the best.  hoever, the idea of a urn containing a secret seems awful :)

Reply:
Agreed - may be able to avoid this recursion (assuming we've correctly enforced this elsewhere).